### ISSUE TITLE
Bluetooth/Broadcomm Driver (USB Dongle)

### ISSUE THREAD

**Issue by ppp0** [cargomedia/puppet-packages#979]
Most likely we will use [USB-Bluetooth dongle](http://plugable.com/2014/06/23/plugable-usb-bluetooth-adapter-solving-hfphsp-profile-issues-on-linux) which works with Ubuntu 15.04 but requires some driver/firmware upgrade to support HFP/HSP correctly.

[Steps](http://plugable.com/2014/06/23/plugable-usb-bluetooth-adapter-solving-hfphsp-profile-issues-on-linux)
```
wget https://s3.amazonaws.com/plugable/bin/fw-0a5c_21e8.hcd
sudo mkdir /lib/firmware/brcm
sudo mv fw-0a5c_21e8.hcd /lib/firmware/brcm/BCM20702A0-0a5c-21e8.hcd
sudo reboot
```

More info:
- https://bugs.launchpad.net/ubuntu/+source/bluez/+bug/962082
- http://askubuntu.com/questions/180437/bluetooth-headset-a2dp-works-hsp-hfp-not-no-sound-no-mic

see https://github.com/cargomedia/puppet-packages-ubuntu/issues/17

Docu:
https://github.com/cargomedia/puppet-cargomedia/blob/master/docu/bulldog/setup.md#install-bluetooth-usb-dongle-firmware

**Comment by njam** [comment]
[ASUS USB-BT400](https://www.digitec.ch/de/s1/product/asus-usb-bt400-usb-20-netzwerkadapter-630793) works fine, exactly the same as the one we have. It's the same Broadcom chip.

For the records: when the above patch is *not* applied (on bulldog1) I could get "Headset" service working, but not A2DP. This is the opposite of the problem described in this ticket.

**Comment by kris-lab** [comment]
That's strange! Are you sure you had sound from microphone + sound in the headset speaker from `audio-test` tool or any other source? 

Without `patch` I could install `headset-service`, set as default but there was no real audio (youtube, test or any)!

Can you confirm it works without patch on `ASUS USB-BT400` (`broadcom` based)?

**Comment by njam** [comment]
I didn't test actual playback. The device just paired in "Headset" mode.

Anyway I think we should use A2DP, because the sound quality is much better. Since the chip is the same, let's just apply the fix and it should be fine. wdyt?

**Comment by kris-lab** [comment]
Ok, wihtout path is is pairing in headset mode as well. But there is no `palyback` and `mic` unfortunately.

Agree, let's use A2DP as `mic` will be wired!

**Comment by njam** [comment]
This file (`/lib/firmware/brcm/BCM20702A0-0a5c-21e8.hcd`) is present only on bulldog3 (but not on bulldog2 and bulldog4).

So it seems it's not needed.

@kris-lab @ppp0 please reopen if you disagree.

**Comment by kris-lab** [comment]
I believe this file was on each machine. Is it possible it was removed with Kernel upgrade?

As far as I can see the problem with BROADCOMM was solved with kernel 3.18. When we started with `bulldog` we used 3.16, but we have 3.19+ now.
